src/libsystemd/sd-journal/journal-def.h

```diff
@@ -118,7 +118,10 @@ struct HashTableObject {
 struct EntryArrayObject {
         ObjectHeader object;
         le64_t next_entry_array_offset;
-        le64_t items[];
+        union {
+                le64_t items[0];
+                le32_t compact[0];
```

Comment:
Why not `le32_t compact[];`? That would be standard c code (and might not result in those false positives).

Reply:
Lennart mentioned this as well but flexible arrays are not allowed in unions in standard C. And wrapping them in an anonymous struct doesn't work because you can't have a struct with only a flexible array member field.